op-deployer/pkg/deployer/pipeline/start_block.go

```diff
@@ -57,7 +57,12 @@ func SetStartBlockGenesisStrategy(env *Env, st *state.State, chainID common.Hash
 	if err != nil {
 		return fmt.Errorf("failed to build L1 developer genesis: %w", err)
 	}
-	thisChainState.StartBlock = devGenesis.ToBlock().Header()
+
+	headerFromRPC, err := state.GetHeaderFromRpc(devGenesis.ToBlock().Number(), env.L1Client.Client())
```

Comment:
This is not correct for the genesis strategy. The header must always be calculated for the genesis strategy because the start block of the L2 in this case is the start block of the L1. There is no L1 data to fetch from.